packages/react-model-ui/src/ModelReducer.tsx

```diff
@@ -0,0 +1,160 @@
+/********************************************************************************
+ * Copyright (c) 2023 CrossBreeze.
+ ********************************************************************************/
+import { CrossModelRoot, EntityAttribute } from '@crossbreeze/protocol';
+
+export interface ModelAction {
+    type: string;
+}
+
+export interface ModelUpdateAction extends ModelAction {
+    type: 'model:update';
+    model: CrossModelRoot;
+}
+
+export interface EntityChangeNameAction extends ModelAction {
+    type: 'entity:change-name';
+    name: string;
+}
+
+export interface EntityChangeDescriptionAction extends ModelAction {
+    type: 'entity:change-description';
+    description: string;
+}
+
+export interface EntityAttributeUpdateAction extends ModelAction {
+    type: 'entity:attribute:update';
+    attributeIdx: number;
+    attribute: EntityAttribute;
+}
+
+export interface EntityAttributeChangeNameAction extends ModelAction {
```

Comment:
It seems that after adding the more generic `'entity:attribute:update` action, the actions for name and description are no longer necessary, only the datatype one and even that one could also be mapped to the generic action, what do you think?

Reply:
We can indeed remove some at some point.

After yesterdays chat I investigated the DataGrid stuff a bit more about whether the received updatedRow object could be spread (I think thats the right term) into the EntityAttribute object. I think this can be done, but only after we refactor the langium language to have the proper id property and name stuff. In the end we need some kind of row identifier which is numeric in the datagrid. And also the datatype drop-down logic seems a bit unnatural to me. Would be nice if it would just be a cell edit operation just like the other fields where the processRowUpdate thing would process the result.

So based on the above thinking I thought, maybe first finalize this PR. Let you work on upgrading all deps, then refactor the langium grammer to have the proper names in the ast nodes (so no name_val anymore). And then refactor this grid implementation a bit.

In summary: I agree but I would like to prospone that till later.